test_agent/python/testagent.py

```diff
@@ -65,31 +60,36 @@
 logger = logging.getLogger("File:Line# Debugger")
 logger.setLevel(logging.DEBUG)
 
+RESPONSE_URI = UUri(ue_id=1, ue_version_major=1, resource_id=0)
+
+sdkname = "python"
+transport_name = "socket"
+
 
 class SocketUListener(UListener):
     def on_receive(self, umsg: UMessage) -> None:
         logger.info("Listener received")
         if umsg.attributes.type == UMessageType.UMESSAGE_TYPE_REQUEST:
-            attributes = UAttributesBuilder.response(
-                umsg.attributes.sink,
-                umsg.attributes.source,
-                UPriority.UPRIORITY_CS4,
-                umsg.attributes.id,
-            ).build()
             any_obj = any_pb2.Any()
             any_obj.Pack(StringValue(value="SuccessRPCResponse"))
-            res_msg = UMessage(
-                attributes=attributes,
-                payload=UPayload(
-                    value=any_obj.SerializeToString(),
-                    format=UPayloadFormat.UPAYLOAD_FORMAT_PROTOBUF_WRAPPED_IN_ANY,
-                ),
+            payload = UPayload(
+                data=any_obj.SerializeToString(),
+                format=UPayloadFormat.UPAYLOAD_FORMAT_PROTOBUF_WRAPPED_IN_ANY,
             )
+            res_msg = UMessageBuilder.response(
+                umsg.attributes.sink,
+                umsg.attributes.source,
+                umsg.attributes.id,
+            ).build_from_upayload(payload)
             transport.send(res_msg)
         else:
             send_to_test_manager(umsg, actioncommands.RESPONSE_ON_RECEIVE)
 
 
+transport = SocketUTransport(RESPONSE_URI)
```

Comment:
RESPONSE_URI should be random per uE execution. Add interface to create transport with unique default URI

Reply:
In 1.5.8 URI is required for transport. In order avoid any issue which could occurrence in case to uE have same default uri this change should be part of 1.5.8

Reply:
fair... okay will do